geode-core/src/main/java/org/apache/geode/internal/tcp/ConnectionTable.java

```diff
@@ -443,32 +434,18 @@ Connection getThreadOwnedConnection(InternalDistributedMember id, long startTime
     if (m == null) {
       // First time for this thread. Create thread local
       m = new HashMap();
-      synchronized (threadConnMaps) {
-        if (closed) {
-          owner.getCancelCriterion().checkCancelInProgress(null);
-          throw new DistributedSystemDisconnectedException("Connection table is closed");
-        }
-        // check for stale references and remove them.
-        for (Iterator it = threadConnMaps.iterator(); it.hasNext();) {
-          Reference r = (Reference) it.next();
-          if (r.get() == null) {
-            it.remove();
-          }
-        }
-        threadConnMaps.add(new WeakReference(m));
+      if (closed) {
```

Comment:
I think this check for closing and canceling was here because of the potential blocking at the now removed synchronized block. The calling method checks this before making the call. I don't think checking a few lines later makes any difference, especially since it can change to closed immediately after this check.

Reply:
nice catch; done